cmd/standalone/Dockerfile

```diff
@@ -22,4 +22,4 @@ COPY ./jaeger-ui-build /go/src/jaeger-ui-build
 COPY ./cmd/standalone/standalone-linux /go/bin/
 COPY ./cmd/standalone/sampling_strategies.json /go/src/config/
 
-CMD ["/go/bin/standalone-linux","--query.static-files=/go/src/jaeger-ui-build/build/","--sampling.strategies-file=/go/src/config/sampling_strategies.json"]
+ENTRYPOINT ["/go/bin/standalone-linux", "--query.static-files=/go/src/jaeger-ui-build/build/", "--sampling.strategies-file=/go/src/config/sampling_strategies.json"]
```

Comment:
While I think the static files would "never" change for the Docker image (or would they?), the strategies file are most likely to change in some scenarios. As it's easier to override a config file than a flag on an entrypoint directive, it would be best to use a configuration file instead of passing values via flags, like:

```dockerfile
COPY ./cmd/standalone/standalone.yaml /go/src/config/
ENTRYPOINT ["/go/bin/standalone-linux", "--config-file=/go/src/config/standalone.yaml"]
```

`./cmd/standalone/standalone.yaml` doesn't exist yet, but could look like this:

```yaml
    query:
      static-files: /go/jaeger-ui/
    sampling:
      strategies-file: /go/src/config/sampling_strategies.json
```

The other Dockerfiles should also follow the same pattern.

And finally, I would personally prefer `/conf/` instead of `/go/src/config/`, and `/bin/` instead of `/go/bin/`, unless there's a specific reason to keep `go` in the path name. 

cc @yurishkuro, @pavolloffay

Reply:
I don't have a strong opinion on directories. 

I have noticed that some images put binaries into `/bin` or `/usr/share` and configuration into `/etc` so in our case `/etc/jaeger`

Reply:
Those images are probably following the LSB (Linux Standard Base), which is a standard for this kind of thing on the Linux world. I'd be OK with that, if we do indeed place the config files at `/etc/jaeger/standalone.yaml` instead of `/etc/jaeger.yaml`.

Reply:
I meant `/etc/jaeger` as a dir for whatever configuration we need.

Reply:
But as I said I don't have a strong opinion. I am not sure what benefits putting binaries into /bin has as we use `from scratch`. One way or another following standard Linux conventions is not a bad thing :).

Reply:
> this also entails a code change in each binary to support pulling config values from a passed YAML file

No, this is already handled by Viper

> As there will now only ever be one argument anyways?

Good point. Imagine that someone just typed `docker run jaegertracing/all-in-one --collector.queue-size=10` . What is final expected command? 

`/go/bin/standalone-linux --config-file=/go/src/config/standalone.yaml --collector.queue-size=10`

or

`/go/bin/standalone-linux --collector.queue-size=10`

@pavolloffay what would Viper do in this case? What happens if the config file has "collector.queue-size=11"?

Reply:
I had interpreted the suggested change as eliminating all other flags, so that passing in `--collector.queue-size=10` instead of specifying it in the config file would be an error.

Also,  overriding the config file will require a volume mount specified in `docker run`, right?

Reply:
The precedence is here https://github.com/spf13/viper#why-viper

we should default to the lowest one - definitely not flag because it overrides everything.

Reply:
> `ENTRYPOINT ["/go/bin/standalone-linux", "--config-file=/go/src/config/standalone.yaml"]`

I am not convinced of the value of this. Example shows only two entries in the config, and I think the ui-assets should be simply built into the binary, while strategies by default has to be empty, there's no point in having it in the config.

Reply:
If there are no options that need to be passed, then there's no reason to have a config file. 

If there are CLI options to be passed (as is the case for the current state of this PR), then it's easier for a consumer to override a config file than override the entrypoint directive.

Reply:
I may be a step behind you here- but doesn't the consumer overriding a config file via an argument to the container require a volume mount? In that case it seems easier to override the entrypoint with additional command line arguments.

Reply:
The embedding of UI assets is done in #918. Once it's merged I suggest we change this one to
```
ENTRYPOINT ["/go/bin/standalone-linux"]
CMD ["--sampling.strategies-file=/go/src/config/sampling_strategies.json"]
```
I now see the value of using the config as the lowest priority default value, but I propose we leave it for another PR, as it won't be such a breaking change as this PR.

Reply:
Or rather:

```dockerfile
ENTRYPOINT ["/go/bin/standalone-linux"]
CMD ["--sampling.strategies-file=/etc/jaeger/sampling_strategies.json"]
```

https://github.com/jaegertracing/jaeger/pull/918/files#diff-da47d3e14c88bf43b29700d0fa2b7a27R22

Reply:
@jkandasa scenarios to test:

- Override the CMD with a custom option (`--log-level`, for instance)
- Override the `/etc/jaeger/sampling_strategies.json` file